utf8proc.h

```diff
@@ -268,166 +308,214 @@ typedef struct utf8proc_property_struct {
 #define UTF8PROC_BOUNDCLASS_LVT     10
 #define UTF8PROC_BOUNDCLASS_REGIONAL_INDICATOR 11
 #define UTF8PROC_BOUNDCLASS_SPACINGMARK 12
+/** @} */
 
+/**
+ *  Array containing the byte lengths of a UTF-8 encoded character based
+ *  on the first byte.
+ */
 DLLEXPORT extern const int8_t utf8proc_utf8class[256];
 
+/**
+ *  Returns the version as a string.
+ */
 DLLEXPORT const char *utf8proc_version(void);
 
-DLLEXPORT const char *utf8proc_errmsg(ssize_t errcode);
-/*
+/**
+ *  Con
  *  Returns a static error string for the given error code.
  */
+DLLEXPORT const char *utf8proc_errmsg(ssize_t errcode);
 
-DLLEXPORT ssize_t utf8proc_iterate(const uint8_t *str, ssize_t strlen, int32_t *dst);
-/*
+/**
  *  Reads a single char from the UTF-8 sequence being pointed to by 'str'.
  *  The maximum number of bytes read is 'strlen', unless 'strlen' is
  *  negative.
+ *
  *  If a valid unicode char could be read, it is stored in the variable
  *  being pointed to by 'dst', otherwise that variable will be set to -1.
  *  In case of success the number of bytes read is returned, otherwise a
  *  negative error code is returned.
  */
+DLLEXPORT ssize_t utf8proc_iterate(const uint8_t *str, ssize_t strlen, int32_t *dst);
 
-DLLEXPORT bool utf8proc_codepoint_valid(int32_t uc);
-/*
- *  Returns 1, if the given unicode code-point is valid, otherwise 0.
+/**
+ *  Check if unicode char is a valid unicode code point.
+ *
+ *  @return 1, if the given unicode code-point is valid, otherwise 0.
  */
+DLLEXPORT bool utf8proc_codepoint_valid(int32_t uc);
 
-DLLEXPORT ssize_t utf8proc_encode_char(int32_t uc, uint8_t *dst);
-/*
+/**
  *  Encodes the unicode char with the code point 'uc' as an UTF-8 string in
  *  the byte array being pointed to by 'dst'. This array has to be at least
  *  4 bytes long.
+ *
  *  In case of success the number of bytes written is returned,
  *  otherwise 0.
+ *
  *  This function does not check if 'uc' is a valid unicode code point.
  */
+DLLEXPORT ssize_t utf8proc_encode_char(int32_t uc, uint8_t *dst);
 
-DLLEXPORT const utf8proc_property_t *utf8proc_get_property(int32_t uc);
-/*
- *  Returns a pointer to a (constant) struct containing information about
- *  the unicode char with the given code point 'uc'.
+/**
+ *  Lookup the properties for a given unicode char.
+ *
+ *  @param uc The unicode character.
+ *
+ *  @return
+ *  A pointer to a (constant) struct containing information about
+ *  the unicode char with the given code point `uc`.
+ *  @par
  *  If the character is not existent a pointer to a special struct is
- *  returned, where 'category' is 0 (UTF8PROC_CATEGORY_CN).
+ *  returned, where `category` is 0 (@ref UTF8PROC_CATEGORY_CN).
  */
+DLLEXPORT const utf8proc_property_t *utf8proc_get_property(int32_t uc);
 
+/** Decompose a unicode char into an array of unicode chars.
+ *
+ *  @param uc the unicode char.
```

Comment:
Let's use "codepoint" consistently (instead of "code point" or "char" or "character")